### PR TITLE
feat-595: Add unit test for sdk package

### PR DESCRIPTION
**Pull request by nightfury1204**
https://github.com/convox/issues-private/issues/595

### CODE REVIEW

File: sdk/compact_test.go [hunk 1]
```diff
@@ -0,0 +1,680 @@
...
+		"convox": "rocks",
+	}
+
+	s := stdapi.New("api", "api")
+	s.Route("POST", fmt.Sprintf("/apps/%s/parameters", app), func(c *stdapi.Context) error {
+		for k, v := range params {
+			require.Equal(t, v, c.Form(k))
+		}
+		return c.RenderOK()
+	})
```
Comment (Twsouza) [inline sdk/compact_test.go:51]: Should you check the `app` variable in the URL path?

File: sdk/compact_test.go [hunk 2]
```diff
@@ -0,0 +1,680 @@
...
+		require.NoError(t, err)
+		b, err := io.ReadAll(fp)
+		require.NoError(t, err)
+		require.Equal(t, "test", string(b))
+
+		return c.RenderJSON(build)
+	})
+
+	testServer(t, s, func(c *sdk.Client) {
+		got, err := c.BuildCreateUpload(app, bytes.NewReader([]byte("test")), structs.BuildCreateOptions{
```
Comment (Twsouza) [inline sdk/compact_test.go:81]: Can you move `bytes.NewReader([]byte("test")` to a variable? e.g. `expectedSource`.

**Review by yuriadams — APPROVED** [review]

File: sdk/compact_test.go [hunk 3]
```diff
@@ -0,0 +1,680 @@
...
+		fp, _, err := c.Request().FormFile("image")
+		require.NoError(t, err)
+		b, err := io.ReadAll(fp)
+		require.NoError(t, err)
+		require.Equal(t, "test", string(b))
+		return c.RenderJSON(build)
+	})
+
+	testServer(t, s, func(c *sdk.Client) {
+		got, err := c.BuildImportMultipart(app, bytes.NewReader([]byte("test")))
```
Comment (Twsouza) [inline sdk/compact_test.go:107]: Move to a variable `expectedForm`?

File: sdk/compact_test.go [hunk 4]
```diff
@@ -0,0 +1,680 @@
...
+		b, err := io.ReadAll(c.Request().Body)
+		require.NoError(t, err)
+		require.Equal(t, "test", string(b))
+		return c.RenderJSON(structs.Object{
+			Url: "url",
+		})
+	})
+
+	testServer(t, s, func(c *sdk.Client) {
+		got, err := c.BuildImportUrl(app, bytes.NewReader([]byte("test")))
```
Comment (Twsouza) [inline sdk/compact_test.go:137]: Move it to a variable `expectedBody`?

File: sdk/compact_test.go [hunk 5]
```diff
@@ -0,0 +1,680 @@
...
+		require.Equal(t, "2", c.Header("Height"))
+		require.Equal(t, "1", c.Header("Width"))
+		_, b, err := c.Websocket().ReadMessage()
+		require.NoError(t, err)
+		require.Equal(t, "cmd", string(b))
+		return c.Websocket().WriteMessage(websocket.TextMessage, []byte("data"+statusCodePrefix+"0"))
+	})
+
+	testServer(t, s, func(c *sdk.Client) {
+		rw := bufio.NewReadWriter(bufio.NewReader(bytes.NewReader([]byte("cmd"))), bufio.NewWriter(&bytes.Buffer{}))
```
Comment (Twsouza) [inline sdk/compact_test.go:373]: Move it to a variable `expectedCommand`?

**Review by Twsouza — APPROVED** [review]
LGTM, just some suggestions.

**Pull request merged**